editoast/src/views/train_schedule.rs

```diff
@@ -358,66 +359,40 @@ async fn simulation(
         })
         .await?;
 
-    Ok(Json(
-        train_simulation(
-            &mut db_pool.get().await?,
-            valkey_client,
-            core_client,
-            train_schedule,
-            &infra,
-            electrical_profile_set_id,
-        )
-        .await?
-        .0,
-    ))
-}
-
-/// Compute simulation of a train schedule
-pub async fn train_simulation(
-    conn: &mut DbConnection,
-    valkey_client: Arc<ValkeyClient>,
-    core: Arc<CoreClient>,
-    train_schedule: TrainSchedule,
-    infra: &Infra,
-    electrical_profile_set_id: Option<i64>,
-) -> Result<(SimulationResponse, PathfindingResult)> {
-    Ok(train_simulation_batch(
-        conn,
+    // Compute simulation of a train schedule
+    let (simulation, _) = train_simulation_batch(
+        &mut db_pool.get().await?,
         valkey_client,
-        core,
+        core_client,
         &[train_schedule],
-        infra,
+        &infra,
         electrical_profile_set_id,
     )
     .await?
     .pop()
-    .unwrap())
+    .unwrap(); // TODO handle unwrap ?
+
+    Ok(Json(simulation))
 }
 
-/// Compute in batch the simulation of a list of train schedule
+/// Compute in batch the simulation of a list of train schedule and consist parameters
 ///
 /// Note: The order of the returned simulations is the same as the order of the train schedules.
-pub async fn train_simulation_batch(
+pub async fn consist_train_simulation_batch(
     conn: &mut DbConnection,
     valkey_client: Arc<ValkeyClient>,
     core: Arc<CoreClient>,
     train_schedules: &[TrainSchedule],
     infra: &Infra,
     electrical_profile_set_id: Option<i64>,
+    physics_consist_parameters: PhysicsConsistParameters,
```

Comment:
This parameter must be set for each train schedule, not globaly.

The `consist_train_simulation_batch` should be the function that actually do the simulation.